### PR TITLE
test(EXE): Add more tests for Wasm memory limit

### PR DESCRIPTION
**Pull request by ulan**

### CODE REVIEW

File: rs/execution_environment/tests/execution_test.rs [hunk 1]
```diff
@@ -2359,3 +2359,212 @@ fn helper_tests_for_illegal_data_buffer_access(env: &StateMachine, canister_id:
...
+            canister_id,
+            "update",
+            wasm().push_bytes(&[1, 2, 3]).append_and_reply().build(),
+        )
+        .unwrap();
+    assert_eq!(result, WasmResult::Reply(vec![1, 2, 3]));
+}
+
+#[test]
+fn set_wasm_memory_limit_to_zero() {
```
Comment (berestovskyy) [inline rs/execution_environment/tests/execution_test.rs:2461]: nit: I'm not sure if we have a test for setting the limit lower than the current canister memory usage and executing a message?

**Review by berestovskyy — APPROVED** [review]

File: rs/execution_environment/tests/execution_test.rs (hunk 1 above)
Comment (ulan) [reply rs/execution_environment/tests/execution_test.rs:2456]: In other cases I wanted to check that execution succeeds. In this case it is going to fail. I added an assertion that it indeed fails.

File: rs/execution_environment/tests/execution_test.rs [hunk 2]
```diff
@@ -2359,3 +2359,212 @@ fn helper_tests_for_illegal_data_buffer_access(env: &StateMachine, canister_id:
...
+
+    let wasm_memory_limit: u64 = env
+        .canister_status(canister_id)
+        .unwrap()
+        .unwrap()
+        .settings()
+        .wasm_memory_limit()
+        .0
+        .try_into()
+        .unwrap();
```
[the remark this answers is not in the document]
Comment (ulan) [reply rs/execution_environment/tests/execution_test.rs]: Done

File: rs/execution_environment/tests/execution_test.rs [hunk 3]
```diff
@@ -2359,3 +2359,212 @@ fn helper_tests_for_illegal_data_buffer_access(env: &StateMachine, canister_id:
         "Should return error if input data is copied to out of bound internal buffer. Instead, it returns unexpected message: {}.", ret_val.description()
     );
 }
+
+#[test]
+fn set_wasm_memory_limit_below_memory_usage() {
+    let subnet_config = SubnetConfig::new(SubnetType::Application);
+    let env = StateMachine::new_with_config(StateMachineConfig::new(
```
[the remark this answers is not in the document]
Comment (ulan) [reply rs/execution_environment/tests/execution_test.rs]: Done

**Review by ulan — COMMENTED** [review]
Thanks for review!

File: rs/execution_environment/tests/execution_test.rs (hunk 1 above)
Comment (berestovskyy) [reply rs/execution_environment/tests/execution_test.rs:2456]: Just to clarify, will it fail regardless of whether new memory is allocated?

Comment (ulan) [reply rs/execution_environment/tests/execution_test.rs:2456]: Yes

**Pull request merged**